lib/xenops_server.ml

```diff
@@ -2717,12 +2717,13 @@ module VBD = struct
   let string_of_id (a, b) = a ^ "." ^ b
 
   let add' x =
-    debug "VBD.add %s %s" (string_of_id x.id) (Jsonrpc.to_string (rpc_of t x)) ;
-    (* Only if the corresponding VM actually exists *)
+    let dom0 = Inventory.lookup Inventory._control_domain_uuid in
```

Comment:
this doesn't change, so could be a global, except we need to be careful in case `xenopsd` is the one creating the entry in the first place in `xenops_xc_main`.
Perhaps we could have a `dom0_uuid ref` in `xenops_utils` that is set by `xenops_xc_main` similar to how `Xenops_utils.root` is set, and a get function that retrieves it from that ref (and asserts false on the None branch).